google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -369,38 +372,54 @@ def publish(self, topic, data, ordering_key="", **attrs):
                 "be sent as text strings."
             )
 
-        # Create the Pub/Sub message object.
-        message = types.PubsubMessage(
-            data=data, ordering_key=ordering_key, attributes=attrs
-        )
+        span_name = "{} publisher".format(topic)
+        span_attributes = {"data": data.decode()}
```

Comment:
What is the argument for doing this? It seems to me that, _best_ case, it results in the duplication of data in the message. But in the _worst_ case it actually leaks sensitive information into observability stack, just like logging the output of a SQL query would.

Reply:
I've also noticed it's a potentially unneeded change after experimenting with alternatives myself. Trace information can be passed into pubsub attributes in the current iteration of the pubsub library using Open Telemetry manual instrumentation as far as I can see for now.